lib/FIR/statements.h

```diff
@@ -585,8 +561,10 @@ class Statement : public SumTypeMixin<ReturnStmt,  //
     return reinterpret_cast<std::size_t>(&static_cast<Statement *>(nullptr)->u);
   }
   static Statement *From(Stmt_impl *stmt) {
-    return reinterpret_cast<Statement *>(
-        reinterpret_cast<char *>(stmt) - Statement::offsetof_impl());
+    char *cp{reinterpret_cast<char *>(stmt)};
+    auto adjust{Statement::offsetof_impl()};
+    CHECK(adjust == 0);
+    return reinterpret_cast<Statement *>(cp - adjust);
```

Comment:
I still find the confusing and unportable. If you have to get from a Stmt_impl back to the Statement that contains it, why not store a pointer to the Statement in the Stmt_impl?